File: bindings/lua/Polycode/SceneLight.lua
require "Polycode/Entity"

class "SceneLight" (Entity)

SceneLight.POINT_LIGHT = 0
SceneLight.SPOT_LIGHT = 1

function SceneLight:SceneLight(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "Entity" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SceneLight(unpack(arg))
	end
end

function SceneLight:getIntensity()
	local retVal =  Polycode.SceneLight_getIntensity(self.__ptr)
	return retVal
end

function SceneLight:setIntensity(newIntensity)
	local retVal = Polycode.SceneLight_setIntensity(self.__ptr, newIntensity)
end

function SceneLight:setAttenuation(constantAttenuation, linearAttenuation, quadraticAttenuation)
	local retVal = Polycode.SceneLight_setAttenuation(self.__ptr, constantAttenuation, linearAttenuation, quadraticAttenuation)
end

function SceneLight:getConstantAttenuation()
	local retVal =  Polycode.SceneLight_getConstantAttenuation(self.__ptr)
	return retVal
end

function SceneLight:getLinearAttenuation()
	local retVal =  Polycode.SceneLight_getLinearAttenuation(self.__ptr)
	return retVal
end

function SceneLight:getQuadraticAttenuation()
	local retVal =  Polycode.SceneLight_getQuadraticAttenuation(self.__ptr)
	return retVal
end

function SceneLight:getType()
	local retVal =  Polycode.SceneLight_getType(self.__ptr)
	return retVal
end

function SceneLight:renderDepthMap(scene)
	local retVal = Polycode.SceneLight_renderDepthMap(self.__ptr, scene.__ptr)
end

function SceneLight:getZBufferTexture()
	local retVal =  Polycode.SceneLight_getZBufferTexture(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Texture"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneLight:setSpecularLightColor(r, g, b, a)
	local retVal = Polycode.SceneLight_setSpecularLightColor(self.__ptr, r, g, b, a)
end

function SceneLight:setDiffuseLightColor(r, g, b, a)
	local retVal = Polycode.SceneLight_setDiffuseLightColor(self.__ptr, r, g, b, a)
end

function SceneLight:setLightColor(r, g, b, a)
	local retVal = Polycode.SceneLight_setLightColor(self.__ptr, r, g, b, a)
end

function SceneLight:setSpotlightProperties(spotlightCutoff, spotlightExponent)
	local retVal = Polycode.SceneLight_setSpotlightProperties(self.__ptr, spotlightCutoff, spotlightExponent)
end

function SceneLight:getSpotlightCutoff()
	local retVal =  Polycode.SceneLight_getSpotlightCutoff(self.__ptr)
	return retVal
end

function SceneLight:getSpotlightExponent()
	local retVal =  Polycode.SceneLight_getSpotlightExponent(self.__ptr)
	return retVal
end

function SceneLight:enableShadows(val, resolution)
	local retVal = Polycode.SceneLight_enableShadows(self.__ptr, val, resolution)
end

function SceneLight:setShadowMapFOV(fov)
	local retVal = Polycode.SceneLight_setShadowMapFOV(self.__ptr, fov)
end

function SceneLight:getShadowMapFOV()
	local retVal =  Polycode.SceneLight_getShadowMapFOV(self.__ptr)
	return retVal
end

function SceneLight:getShadowMapResolution()
	local retVal =  Polycode.SceneLight_getShadowMapResolution(self.__ptr)
	return retVal
end

function SceneLight:areShadowsEnabled()
	local retVal =  Polycode.SceneLight_areShadowsEnabled(self.__ptr)
	return retVal
end

function SceneLight:getLightType()
	local retVal =  Polycode.SceneLight_getLightType(self.__ptr)
	return retVal
end

function SceneLight:setLightImportance(newImportance)
	local retVal = Polycode.SceneLight_setLightImportance(self.__ptr, newImportance)
end

function SceneLight:getLightImportance()
	local retVal =  Polycode.SceneLight_getLightImportance(self.__ptr)
	return retVal
end

function SceneLight:setLightType(lightType)
	local retVal = Polycode.SceneLight_setLightType(self.__ptr, lightType)
end

function SceneLight:Clone(deepClone, ignoreEditorOnly)
	local retVal = Polycode.SceneLight_Clone(self.__ptr, deepClone, ignoreEditorOnly)
	if retVal == nil then return nil end
	local __c = _G["Entity"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneLight:applyClone(clone, deepClone, ignoreEditorOnly)
	local retVal = Polycode.SceneLight_applyClone(self.__ptr, clone.__ptr, deepClone, ignoreEditorOnly)
end

function SceneLight:getParentScene()
	local retVal =  Polycode.SceneLight_getParentScene(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Scene"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneLight:setParentScene(scene)
	local retVal = Polycode.SceneLight_setParentScene(self.__ptr, scene.__ptr)
end

function SceneLight:getSpotlightCamera()
	local retVal =  Polycode.SceneLight_getSpotlightCamera(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Camera"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneLight:getLightInfo()
	local retVal =  Polycode.SceneLight_getLightInfo(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["LightInfo"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneLight:__delete()
	if self then Polycode.delete_SceneLight(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneLight.js
require('Polycode/Entity')

function SceneLight(type,parentScene,intensity,constantAttenuation,linearAttenuation,quadraticAttenuation) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneLight(type,parentScene,intensity,constantAttenuation,linearAttenuation,quadraticAttenuation)
	}
}

SceneLight.POINT_LIGHT = 0
SceneLight.SPOT_LIGHT = 1

SceneLight.prototype = Object.create(Entity.prototype)


SceneLight.prototype.getIntensity = function() {
	return Polycode.SceneLight_getIntensity(this.__ptr)
}

SceneLight.prototype.setIntensity = function(newIntensity) {
	Polycode.SceneLight_setIntensity(this.__ptr, newIntensity)
}

SceneLight.prototype.setAttenuation = function(constantAttenuation,linearAttenuation,quadraticAttenuation) {
	Polycode.SceneLight_setAttenuation(this.__ptr, constantAttenuation, linearAttenuation, quadraticAttenuation)
}

SceneLight.prototype.getConstantAttenuation = function() {
	return Polycode.SceneLight_getConstantAttenuation(this.__ptr)
}

SceneLight.prototype.getLinearAttenuation = function() {
	return Polycode.SceneLight_getLinearAttenuation(this.__ptr)
}

SceneLight.prototype.getQuadraticAttenuation = function() {
	return Polycode.SceneLight_getQuadraticAttenuation(this.__ptr)
}

SceneLight.prototype.getType = function() {
	return Polycode.SceneLight_getType(this.__ptr)
}

SceneLight.prototype.renderDepthMap = function(scene) {
	Polycode.SceneLight_renderDepthMap(this.__ptr, scene.__ptr)
}

SceneLight.prototype.getZBufferTexture = function() {
	var retVal = new Texture()
	retVal.__ptr = Polycode.SceneLight_getZBufferTexture(this.__ptr)
	return retVal
}

SceneLight.prototype.setSpecularLightColor = function(r,g,b,a) {
	Polycode.SceneLight_setSpecularLightColor(this.__ptr, r, g, b, a)
}

SceneLight.prototype.setDiffuseLightColor = function(r,g,b,a) {
	Polycode.SceneLight_setDiffuseLightColor(this.__ptr, r, g, b, a)
}

SceneLight.prototype.setLightColor = function(r,g,b,a) {
	Polycode.SceneLight_setLightColor(this.__ptr, r, g, b, a)
}

SceneLight.prototype.setSpotlightProperties = function(spotlightCutoff,spotlightExponent) {
	Polycode.SceneLight_setSpotlightProperties(this.__ptr, spotlightCutoff, spotlightExponent)
}

SceneLight.prototype.getSpotlightCutoff = function() {
	return Polycode.SceneLight_getSpotlightCutoff(this.__ptr)
}

SceneLight.prototype.getSpotlightExponent = function() {
	return Polycode.SceneLight_getSpotlightExponent(this.__ptr)
}

SceneLight.prototype.enableShadows = function(val,resolution) {
	Polycode.SceneLight_enableShadows(this.__ptr, val, resolution)
}

SceneLight.prototype.setShadowMapFOV = function(fov) {
	Polycode.SceneLight_setShadowMapFOV(this.__ptr, fov)
}

SceneLight.prototype.getShadowMapFOV = function() {
	return Polycode.SceneLight_getShadowMapFOV(this.__ptr)
}

SceneLight.prototype.getShadowMapResolution = function() {
	return Polycode.SceneLight_getShadowMapResolution(this.__ptr)
}

SceneLight.prototype.areShadowsEnabled = function() {
	return Polycode.SceneLight_areShadowsEnabled(this.__ptr)
}

SceneLight.prototype.getLightType = function() {
	return Polycode.SceneLight_getLightType(this.__ptr)
}

SceneLight.prototype.setLightImportance = function(newImportance) {
	Polycode.SceneLight_setLightImportance(this.__ptr, newImportance)
}

SceneLight.prototype.getLightImportance = function() {
	return Polycode.SceneLight_getLightImportance(this.__ptr)
}

SceneLight.prototype.setLightType = function(lightType) {
	Polycode.SceneLight_setLightType(this.__ptr, lightType)
}

SceneLight.prototype.Clone = function(deepClone,ignoreEditorOnly) {
	var retVal = new Entity()
	retVal.__ptr = Polycode.SceneLight_Clone(this.__ptr, deepClone, ignoreEditorOnly)
	return retVal
}

SceneLight.prototype.applyClone = function(clone,deepClone,ignoreEditorOnly) {
	Polycode.SceneLight_applyClone(this.__ptr, clone.__ptr, deepClone, ignoreEditorOnly)
}

SceneLight.prototype.getParentScene = function() {
	var retVal = new Scene()
	retVal.__ptr = Polycode.SceneLight_getParentScene(this.__ptr)
	return retVal
}

SceneLight.prototype.setParentScene = function(scene) {
	Polycode.SceneLight_setParentScene(this.__ptr, scene.__ptr)
}

SceneLight.prototype.getSpotlightCamera = function() {
	var retVal = new Camera()
	retVal.__ptr = Polycode.SceneLight_getSpotlightCamera(this.__ptr)
	return retVal
}

SceneLight.prototype.getLightInfo = function() {
	var retVal = new LightInfo()
	retVal.__ptr = Polycode.SceneLight_getLightInfo(this.__ptr)
	return retVal
}
